bin/bootstrax

```diff
@@ -683,19 +682,26 @@ def _delete_data(rd, path, data_type):
     if data_type == 'live' and not args.delete_live and args.production:
         raise ValueError('Unsafe operation. Trying to delete live data!')
     log.info(f'Deleting data at {path}')
-    shutil.rmtree(path)
+    if os.path.exists(path):
+        shutil.rmtree(path)
     log.info(f'deleting {path} finished')
     # Remove the data location from the rundoc and append it to the 'deleted_data' entries
     if not os.path.exists(path):
         log.info('changing data field in rundoc')
-        ddoc = run_coll.find_one({'_id': rd['_id'], 'data.type': data_type})
-        del ddoc['location']
+        for ddoc in rd['data']:
+            if ddoc['type'] == data_type:
+                break
+        for k in ddoc.copy().keys():
+            if k in ['location', 'meta', 'protocol']:
+                ddoc.pop(k)
+
         ddoc.update({'at': now(), 'by': hostname})
+        log.info(f'update with {ddoc}')
         run_coll.update_one({'_id': rd['_id']},
                             {"$addToSet": {'deleted_data': ddoc},
-                             "$pull" : {"data" :
-                                    {"data.type" : data_type,
-                                     "data.host" : hostname}}})
+                             "$pull": {"data":
+                                           {"type": data_type,
+                                            "host": {'$in':['daq', hostname]}}}})
```

Comment:
What about `ddoc["host"]`?

Reply:
I contemplated that too but I want to make sure we only delete stuff that is either from the daq or this machine. What if for some reason we are deleting data that is also e.g. on dali, we should never touch those, bootstrax may only say something about data that is either from itself or from the readers.

Reply:
Good point. We can't trust that data won't have already been shipped out when we call this routine.